lib/index.js

```diff
@@ -34,27 +17,14 @@ module.exports = (referenceDir, config,
 	});
 	let browsers = browserslist.findConfig(referenceDir) || {};
 
-	let plugins = Object.assign({}, DEFAULTS.plugins, config.plugins);
-	let buckets = {
-		static: [],
-		scripts: [],
-		styles: [],
-		markup: []
-	};
-	Object.keys(plugins).forEach(type => {
-		let pluginConfig = config[type];
-		if(!pluginConfig) {
-			return;
-		}
-
-		let { bucket, plugin } = plugins[type];
-		if(!plugin.call) {
-			plugin = loadExtension(plugin, "ERROR: missing plugin");
-		}
-		let build = plugin(pluginConfig, assetManager,
-				{ browsers, sourcemaps, compact });
-		buckets[bucket].push(build);
-	});
+	let plugins = pluginsByBucket(config);
+	// initialize plugins with corresponding configuration
+	let buckets = Object.keys(plugins).reduce((memo, bucket) => {
+		memo[bucket] = plugins[bucket].map(({ plugin, config }) => {
+			return plugin(config, assetManager, { browsers, sourcemaps, compact });
+		});
+		return memo;
+	}, {});
```

Comment:
It's not exactly obvious what the code's doing here, exactly, but I'm not quite sure how that could be improved (even considering https://twitter.com/jaffathecake/status/1213077702300852224). Not a big deal either way though.

Reply:
TBQH, I don't have a problem with this code (or reduce, for that matter)